### PR TITLE
Added list of contributors for v2.3

### PR DESCRIPTION
**Pull request by yair100**
- Added a list of contributors who worked on v2.3. This list isn't complete, but we should merge it with `main` so that the additional strings can be translated.
- Switched to tab spaces

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [x] Tested the changes for accessibility

**Screenshots (optional)**
image

### CODE REVIEW

**Comment by onein528** [comment]
Is there an option to use the GitHub API for contributions? If so, I can help.
However, the GitHub REST API v3 (there is no contribution API in GitHub GraphQL API v4) is a RESTful API and has an N+1 problem(Redundancy problem), which can cause extremely delays if there are many contributors.

The default is 30:
https://api.github.com/repos/files-community/Files/contributors

Specify quantity to 100:
https://api.github.com/repos/files-community/Files/contributors?per_page=100

As I said previously, very slow but no need to change hard coded xaml.

**Comment by yaichenbaum** [comment]
@onein528 that would be great! The intended design is to only show a list of contributors who were involved since the latest update, is this something the API can support?

**Comment by onein528** [comment]
I don't think so. When getting the list of commits, the `Since` parameter is present, but there is no such param in listing repo contributors.

In that case, there is only one way to list the authors of the commits after the specified date(since param)(*1). Due to the duplication and of course the high number of commits, running that query every time you load the About page will impact performance. Therefore, I think it is a good idea to create a json somewhere and update the list regularly after the version upgrade.

(*1) How about I make the console app to list contributors, run it and copy and paste it before the version upgrade? (I dont know well if I can. but I can try 😅)

**Comment by onein528** [comment]
Wherこぴーしましｔ

**Comment by yaichenbaum** [comment]
> (*1) How about I make the console app to list contributors, run it and copy and paste it before the version upgrade? (I dont know well if I can. but I can try 😅)

That's better than trying to manually create the list 😅 
Keep me updated 👍

**Comment by 0x5bfa** [comment]
@yair100 Can you specify specific since date or commit?

**Comment by onein528** [comment]
There is `since`, `until`, `sha(commit hash/oid)`, and more param:

image

**Comment by yair100** [comment]
> @yair100 Can you specify specific since date or commit?

Since commit is good

**Comment by onein528** [comment]
yep. I think so. but need to specify. Can you give me date what you want to start counting?

**Comment by onein528** [comment]
Nice!

**Comment by 0x5bfa** [comment]
Source code in C#:

```cs
using Octokit;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net.Http;
using System.Threading;
using System.Threading.Tasks;

namespace FilesContributors
{
    class App
    {
        public static GitHubClient Client { get; private set; } = new GitHubClient(new ProductHeaderValue("FluentHub"));

        static async Task Main()
        {
            Console.WriteLine("********************************************************************************");
            Console.WriteLine("*                                                                              *");
            Console.WriteLine("*  FilesContributors                                                           *");
            Console.WriteLine("*  ------------------------------                                              *");
            Console.WriteLine("*   Licensed under the MIT License.                                            *");
            Console.WriteLine("*   Copyright (c) 2022 files-community                                         *");
            Console.WriteLine("*                                                                              *");
            Console.WriteLine("********************************************************************************");
            Console.WriteLine("");

            Client.Credentials = new Credentials("token");

            var list = Normalize(await GetContributors(1));

            Console.WriteLine("Contributors since 36d5996");
            Console.WriteLine("--------------------------------------------------------------------------------");
            foreach (var item in list)
            {
                Console.WriteLine($"{item}");
            }
        }

        static async Task<List<GitHubCommit>> GetContributors(int pageCount)
        {
            CommitRequest commitRequest = new()
            {
                Sha = "36d5996baec91c0d571bf96637f949b8706a12f4",
            };

            ApiOptions apiOptins = new()
            {
                PageCount = pageCount,
                PageSize = 100,
                StartPage = 1, // 1-based
            };


            var commits = await Client.Repository.Commit.GetAll("files-community", "Files", commitRequest, apiOptins);

            return commits.ToList();
        }

        static List<string> Normalize(List<GitHubCommit> commits)
        {
            List<string> commitsHumanized = new();

            foreach (var commit in commits)
            {
                string commitHumanized = $"<HyperlinkButton Content=\"{commit.Author?.Login}\" NavigateUri=\"{commit.Author?.HtmlUrl}\" />";

                commitsHumanized.Add(commitHumanized);
            }

            return commitsHumanized.Distinct().ToList();
        }
    }
}
```

Result:
```cmd
********************************************************************************
*                                                                              *
*  FilesContributors                                                           *
*  ------------------------------                                              *
*   Licensed under the MIT License.                                            *
*   Copyright (c) 2022 files-community                                         *
*                                                                              *
********************************************************************************

Contributors since 36d5996
--------------------------------------------------------------------------------
<HyperlinkButton Content="gave92" NavigateUri="https://github.com/gave92" />
<HyperlinkButton Content="R3voA3" NavigateUri="https://github.com/R3voA3" />
<HyperlinkButton Content="cinqmilleans" NavigateUri="https://github.com/cinqmilleans" />
<HyperlinkButton Content="yair100" NavigateUri="https://github.com/yair100" />
<HyperlinkButton Content="puppetsw" NavigateUri="https://github.com/puppetsw" />
<HyperlinkButton Content="jiejasonliu" NavigateUri="https://github.com/jiejasonliu" />
<HyperlinkButton Content="shiazza" NavigateUri="https://github.com/shiazza" />
<HyperlinkButton Content="szabolcstarnai" NavigateUri="https://github.com/szabolcstarnai" />
<HyperlinkButton Content="onein528" NavigateUri="https://github.com/onein528" />
<HyperlinkButton Content="manfromarce" NavigateUri="https://github.com/manfromarce" />
<HyperlinkButton Content="devovercome" NavigateUri="https://github.com/devovercome" />
<HyperlinkButton Content="SudaCat" NavigateUri="https://github.com/SudaCat" />
<HyperlinkButton Content="dependabot[bot]" NavigateUri="https://github.com/apps/dependabot" />
<HyperlinkButton Content="bebeyaco" NavigateUri="https://github.com/bebeyaco" />
<HyperlinkButton Content="lukeblevins" NavigateUri="https://github.com/lukeblevins" />
<HyperlinkButton Content="d2dyno1" NavigateUri="https://github.com/d2dyno1" />
<HyperlinkButton Content="somnisomni" NavigateUri="https://github.com/somnisomni" />
<HyperlinkButton Content="xlifi" NavigateUri="https://github.com/xlifi" />
<HyperlinkButton Content="konradsikorski" NavigateUri="https://github.com/konradsikorski" />
<HyperlinkButton Content="oleklukasiewicz" NavigateUri="https://github.com/oleklukasiewicz" />
<HyperlinkButton Content="mafra99" NavigateUri="https://github.com/mafra99" />
<HyperlinkButton Content="THEBOSSMAGNUS" NavigateUri="https://github.com/THEBOSSMAGNUS" />
<HyperlinkButton Content="elvistkf" NavigateUri="https://github.com/elvistkf" />
<HyperlinkButton Content="tsakalerisk" NavigateUri="https://github.com/tsakalerisk" />
<HyperlinkButton Content="michalmrowiec" NavigateUri="https://github.com/michalmrowiec" />
<HyperlinkButton Content="oxixes" NavigateUri="https://github.com/oxixes" />
<HyperlinkButton Content="Daxxxis" NavigateUri="https://github.com/Daxxxis" />
<HyperlinkButton Content="ahmed605" NavigateUri="https://github.com/ahmed605" />
<HyperlinkButton Content="pklion" NavigateUri="https://github.com/pklion" />
<HyperlinkButton Content="Wtcheesecakes" NavigateUri="https://github.com/Wtcheesecakes" />
<HyperlinkButton Content="slorixsh" NavigateUri="https://github.com/slorixsh" />
<HyperlinkButton Content="FilopateerMaged" NavigateUri="https://github.com/FilopateerMaged" />
<HyperlinkButton Content="acesyde" NavigateUri="https://github.com/acesyde" />
```

**Comment by onein528** [comment]
you have to change tabs to 4 spaces maybe.

**Comment by yaichenbaum** [comment]
> you have to change tabs to 4 spaces maybe.

We're transitioning over to tabs to be consistent with the c# files.

File: src/Files.Uwp/Views/SettingsPages/About.xaml [hunk 1]
```diff
@@ -1,248 +1,291 @@
...
+						<HyperlinkButton Content="shiazza" NavigateUri="https://github.com/shiazza" />
+						<HyperlinkButton Content="szabolcstarnai" NavigateUri="https://github.com/szabolcstarnai" />
+						<HyperlinkButton Content="onein528" NavigateUri="https://github.com/onein528" />
+						<HyperlinkButton Content="manfromarce" NavigateUri="https://github.com/manfromarce" />
+						<HyperlinkButton Content="devovercome" NavigateUri="https://github.com/devovercome" />
+						<HyperlinkButton Content="SudaCat" NavigateUri="https://github.com/SudaCat" />
+						<HyperlinkButton Content="dependabot[bot]" NavigateUri="https://github.com/apps/dependabot" />
+						<HyperlinkButton Content="bebeyaco" NavigateUri="https://github.com/bebeyaco" />
+						<HyperlinkButton Content="lukeblevins" NavigateUri="https://github.com/lukeblevins" />
+						<HyperlinkButton Content="d2dyno1" NavigateUri="https://github.com/d2dyno1" />
```
Comment (d2dyno1) [inline src/Files.Uwp/Views/SettingsPages/About.xaml:249]:
```suggestion
						<HyperlinkButton Content="d2dyno" NavigateUri="https://github.com/d2dyno1" />
```

**Comment by d2dyno1** [comment]
Perhaps we could pull the contributors list from an API like Octokit?

File: src/Files.Uwp/Views/SettingsPages/About.xaml [hunk 2]
```diff
@@ -1,248 +1,291 @@
...
+	xmlns:helpers="using:Files.Uwp.Helpers"
+	xmlns:i="using:Microsoft.Xaml.Interactivity"
+	xmlns:ic="using:Microsoft.Xaml.Interactions.Core"
+	xmlns:local="using:Files.Uwp.UserControls.Settings"
+	xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+	mc:Ignorable="d">
+	<Page.Resources>
+		<ResourceDictionary>
+			<ResourceDictionary.MergedDictionaries>
+				<ResourceDictionary Source="/ResourceDictionaries/RightAlignedToggleSwitchStyle.xaml" />
```
Comment (d2dyno1) [inline src/Files.Uwp/Views/SettingsPages/About.xaml:16]: I don't think that's needed really. I found out that you could just set MinWidth="0" and it'll be aligned.

**Comment by onein528** [comment]
@d2dyno1 Why changed your name?

d2dyno is `Name`
d2dyno1 is `Login` in Octokit

me:
U+5BFA is `Name`
onein528 is `Login`

image

If your name is to be `Name`, then the text for everyone else must also be changed. Also, since `Name` can be easily changed and up to 40 characters can be added, it is usually a long name and may not fit in the HyperlinkButton's Content.

File: src/Files.Uwp/Views/SettingsPages/About.xaml (hunk 2 above)
Comment (yaichenbaum) [reply src/Files.Uwp/Views/SettingsPages/About.xaml:16]: I'm keen to go through and remove these styles at some point but it's not related to this PR as this line was already there before.

**Review by d2dyno1 — APPROVED** [review]

**Comment by d2dyno1** [comment]
@onein528 thanks! I decided to use my display nick in this case though 🙂

(I accidentaly edited your comment which was supposed to be an answer to yours. Ah github on mobile)

**Pull request merged**